apps/web/src/app/[lang]/dho/[id]/agreements/page.tsx

```diff
@@ -1,20 +1,34 @@
+'use client';
 import { Locale } from "@hypha-platform/i18n";
 import { Tabs, TabsList, TabsTrigger } from "@hypha-platform/ui/server";
 import Link from "next/link";
 import { ListDiscussions, ListProposals, ListAgreements } from '@hypha-platform/epics';
-import { Text } from "@radix-ui/themes";
-import { Carousel, CarouselContent, CarouselItem } from "@hypha-platform/ui";
-import { CardOrganisation } from "@hypha-platform/epics";
-import { getDhoPathAgreements } from "../constants";
-import { getAccessToken, getDaoList } from '@hypha-platform/graphql/rsc';
+import { listProposalsData, listDiscussionsData, listAgreementsData } from "@hypha-platform/ui-utils";
+import { useState } from "react";
 
 type PageProps = {
   params: { lang: Locale; id: string };
 };
 
-export default async function AgreementsPage({ params: { lang, id } }: PageProps) {
-  const newtoken = await getAccessToken();
-  const daos = await getDaoList({ token: newtoken.accessJWT });
+export default function AgreementsPage({ params: { lang, id } }: PageProps) {
+
+  const [proposals, setProposals] = useState(listProposalsData.proposals);
+  const [discussions, setDiscussions] = useState(listDiscussionsData.discussions);
+  const [agreements, setAgreements] = useState(listAgreementsData.agreements);
+
+  const loadMoreProposals = () => {
+    const newProposals = listProposalsData.newProposals;
+    setProposals(prevProposals => [...prevProposals, ...newProposals]);
+  }
```

Comment:
Let's create a `useProposals` hook in `packages/epics/src/proposals/hooks/use-proposals.ts`. Next to that file you can put a `use-proposals.mock.ts` to store the mock data.

Reply:
These specialised hooks should encapsulate the loading and caching functionality using [swr](https://www.npmjs.com/package/swr) or [react-query](https://www.npmjs.com/package/react-query). Let's have a discussion about this.